src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -266,22 +347,53 @@ public void Save(ModelSaveContext ctx)
             _host.AssertValue(ctx);
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
+            ctx.Writer.Write(IsFrozen ? 1 : 0);
+
             // *** Binary format ***
+            // int: indicator for frozen models
             // stream: tensorFlow model.
             // int: number of input columns
             // for each input column
             //   int: id of int column name
             // int: number of output columns
             // for each output column
             //   int: id of output column name
+            if (IsFrozen)
+            {
+                var buffer = new TFBuffer();
+                Session.Graph.ToGraphDef(buffer);
 
-            var buffer = new TFBuffer();
-            Session.Graph.ToGraphDef(buffer);
-
-            ctx.SaveBinaryStream("TFModel", w =>
+                ctx.SaveBinaryStream("TFModel", w =>
+                {
+                    w.WriteByteArray(buffer.ToArray());
+                });
+            }
+            else
             {
-                w.WriteByteArray(buffer.ToArray());
-            });
+                var savedModel = new Dictionary<string, byte[]>();
+                if (!string.IsNullOrEmpty(ModelPath))
```

Comment:
>ModelPath [](start = 42, length = 9)

When this `ModelPath` wont be `NullOrEmpty`? I assume that only for the first `save`. After every subsequent `save` it will be null and dictionary will be used instead, correct?

Can we always load it in constructor instead? It will bit simplify the logic.

Reply:
Also, `ModelPath` directory wont be required after the constructor is done.

---
In reply to: [217478783](https://github.com/dotnet/machinelearning/pull/853#discussion_r217478783) [](ancestors = 217478783)

Reply:
Correct. This indeed makes logic simpler. Thanks.

---
In reply to: [217479179](https://github.com/dotnet/machinelearning/pull/853#discussion_r217479179) [](ancestors = 217479179,217478783)